src/solvers/refinement/string_refinement.cpp

```diff
@@ -776,7 +776,7 @@ decision_proceduret::resultt string_refinementt::dec_solve()
   while((loop_bound_--)>0)
   {
     dependencies.clean_cache();
-    const decision_proceduret::resultt res=supert::dec_solve();
+    res = supert::dec_solve();
```

Comment:
I'm not a big fan of this change - makes it hard to see that this is constant throughout the loop and not dependent on it's outside loop state. A nice fix would of course be to introduce some functions or something, but perhaps an easier immediate fix for the shadowing would be call this  `loop_res` and the other one `initial_res`

Reply:
Renaming done as suggested (using `initial_result`, `refined_result`).